### ISSUE TITLE
NaN or Inf found in input tensor.

### ISSUE THREAD

**Issue by xx210** [FanChiMao/CMFNet#3]
I encounter this problem when i train.
NaN or Inf found in input tensor.
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 861/861 [06:36<00:00,  2.17it/s]
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
[epoch 8 PSNR: nan --- best_epoch 3 Best_PSNR 23.9811]
[epoch 8 SSIM: nan --- best_epoch 5 Best_SSIM 0.8553]

**Comment by FanChiMao** [comment]
The above mentioned problem you faced, which is gradient explosion.
Because the value of loss is represented as the gradient, which is calculated by the differential.
This is standard phenomenon of gradient explosion. 
Usually happen on `the training data is insufficient` or `the architecture of training model is too complex`,
the gradient vanish is also caused by the same reason but the loss value will be displayed as zero.
There are some common solutions to solve the gradient problem:
  1. Decrease the initial learning rate to slow down the convergence time.
  2. Decrease the parameters of model. (Like the initial feature channels from 96 to 64)
  3. Increase the training data.

In your case, I noticed that your training data has only 861 (raindrop) images, it's not enough.
Please see [**Train**](https://github.com/FanChiMao/CMFNet#train) part, subsection **Details of Hyperparameters** in README.md and [**preprocess**](https://github.com/FanChiMao/CMFNet/tree/main/datasets#preprocess) in dataset README.md to increase your training data.
